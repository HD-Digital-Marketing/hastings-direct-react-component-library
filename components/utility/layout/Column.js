import React from 'react'
import PropTypes from 'prop-types'
import './grid.scss'
import classNames from "classnames";

const Column = props => {

    const {
        className = '',
        xs,
        sm,
        md,
        lg,
        xl,
        offset,
        children
    } = props

    const colClass = classNames(className, {
        [`g-col-xs-${xs}`]: xs,
        [`g-col-sm-${sm}`]: sm,
        [`g-col-md-${md}`]: md,
        [`g-col-lg-${lg}`]: lg,
        [`g-col-xl-${xl}`]: xl
    })

    let offsetClass = '';

    if (offset) {
        offsetClass = classNames({
            [`g-col-xs-offset-${offset.xs}`]: offset.xs > -1,
            [`g-col-sm-offset-${offset.sm}`]: offset.sm > -1,
            [`g-col-md-offset-${offset.md}`]: offset.md > -1,
            [`g-col-lg-offset-${offset.lg}`]: offset.lg > -1,
            [`g-col-xl-offset-${offset.xl}`]: offset.xl > -1
        })
    }

    return(
        <div className={`${colClass} ${offsetClass}`}>{children}</div>
    )
}

Column.propTypes = {
    className: PropTypes.string,
    xs: PropTypes.number,
    sm: PropTypes.number,
    md: PropTypes.number,
    lg: PropTypes.number,
    xl: PropTypes.number,
    offset: PropTypes.object,
    children: PropTypes.node
}

export default Column
