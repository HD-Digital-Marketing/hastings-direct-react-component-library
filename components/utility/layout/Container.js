import React from 'react'
import classNames from 'classnames'
import PropTypes from 'prop-types'
import './grid.scss'

const Container = props => {

    const {
        fluid = false,
        className = '',
        children
    } = props

    const containerClass = classNames('g-container', className, {
        'g-container--fluid': fluid
    })

    return(
        <div className={containerClass}>{children}</div>
    )
}

Container.propTypes = {
    fluid: PropTypes.bool,
    className: PropTypes.string,
    children: PropTypes.node
}

export default Container
