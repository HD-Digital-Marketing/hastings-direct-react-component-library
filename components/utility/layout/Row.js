import React from 'react'
import classNames from 'classnames'
import PropTypes from 'prop-types'
import './grid.scss'

const Row = props => {

    const {
        className = '',
        children
    } = props

    const rowClass = classNames('g-row', className)

    return(
        <div className={rowClass}>{children}</div>
    )
}

Row.propTypes = {
    className: PropTypes.string,
    children: PropTypes.node
}

export default Row
