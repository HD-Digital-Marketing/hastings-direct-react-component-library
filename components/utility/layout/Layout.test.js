import React from 'react'
import renderer from 'react-test-renderer'
import Container from './Container';
import Row from './Row';
import Column from './Column';

describe('Check the grid renders', () => {
    test('it should match the snapshot', () => {

        const tree = renderer
            .create(
                <Container fluid={true}>
                    <Row>
                        <Column xs={12} sm={10} md={8} lg={6} xl={4}>
                            This is a column
                        </Column>
                    </Row>
                </Container>
            )
            .toJSON()
        expect(tree).toMatchSnapshot()

    })
})