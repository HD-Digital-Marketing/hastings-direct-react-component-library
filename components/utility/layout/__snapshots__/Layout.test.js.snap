// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check the grid renders it should match the snapshot 1`] = `
<div
  className="g-container g-container--fluid"
>
  <div
    className="g-row"
  >
    <div
      className="g-col-xs-12 g-col-sm-10 g-col-md-8 g-col-lg-6 g-col-xl-4 "
    >
      This is a column
    </div>
  </div>
</div>
`;
